components/src/core/drawer/drawer.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useCallback } from 'react';
+import { View, StyleSheet } from 'react-native';
+import { DrawerInheritableProps, inheritDrawerProps } from './inheritable-types';
+import * as Colors from '@pxblue/colors';
+import MatIcon from 'react-native-vector-icons/MaterialIcons';
+
+const styles = StyleSheet.create({
+    container: {
+        backgroundColor: Colors.white[50],
+        zIndex: 2,
+        flex: 1,
+        height: '100%',
+    },
+});
+
+export const Drawer: React.FC<DrawerInheritableProps> = (props) => {
+    // Set theme-related default props here.
+    const activeItemBackgroundColor = Colors.blue[50];
+    const findChildByType = useCallback(
+        (type: string): JSX.Element[] =>
+            React.Children.map(props.children, (child: any) => {
+                if (child && child.type) {
+                    const name = child.type.displayName;
+                    if (name && name.includes(type)) {
+                        return child;
+                    }
+                }
+            }) || [],
+        [props]
+    );
+
+    const getSectionByDisplayName = useCallback(
+        (displayName: string, inherit = false): JSX.Element[] =>
+            findChildByType(displayName)
+                .slice(0, 1)
+                .map((child) => {
+                    let inheritableProps = {};
+                    if (inherit) {
+                        inheritableProps = inheritDrawerProps({ ...props, activeItemBackgroundColor }, child.props);
+                    }
+                    return React.cloneElement(child, inheritableProps);
+                }),
+        [props]
+    );
+
+    return (
+        <View style={styles.container}>
+            {getSectionByDisplayName('DrawerHeader')}
+            {getSectionByDisplayName('DrawerSubheader')}
+            {getSectionByDisplayName('DrawerBody', true)}
```

Comment:
why is drawerBody the only one with inherit=true?

Reply:
I suppose the other pieces don't actually care about any of the drawer props like they did in react because we don't have the extra behavior of the persistent drawer?

Reply:
Yep that's why.  The `inheritableProps` have this lineage: 
Drawer -> DrawerBody -> DrawerNavGroup -> DrawerNavItem

Since header, subheader, footer dont fall into the lineage, they don't need to inherit anything.